### PR TITLE
Improve ORC writer benchmark with nvbench

### PR DESCRIPTION
**Pull request by PointKernel**
## Description

Issue #10941 

Changes involved in this PR:

- Rewrites the ORC writer benchmarks with nvbench
- Cleans up the existing implementations by moving nvbench macros to a new `nvbench_helpers.hpp` header
- Fixes two small issues in `orc_writer_chunks` benchmarks 
  - Set `state` target stream
  - Set up memory pool
- Removes unused headers
- Gets rid of `cudf_io` aliases
- Fixes a bug in the current ORC writer benchmark: calculating the size of one single file

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by hyperbolic2346 — APPROVED** [review]
This is great work, thanks!

File: cpp/benchmarks/io/nvbench_helpers.hpp [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+#pragma once
+
+#include <benchmarks/common/generate_input.hpp>
+#include <benchmarks/io/cuio_common.hpp>
+
+#include <cudf/io/types.hpp>
+#include <cudf/types.hpp>
+
+#include <nvbench/nvbench.cuh>
```
Comment (ttnghia) [inline cpp/benchmarks/io/nvbench_helpers.hpp:25]: I think this header should come first.

File: cpp/benchmarks/io/nvbench_helpers.hpp [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+enum class data_type : int32_t {
+  INTEGRAL        = static_cast<int32_t>(type_group_id::INTEGRAL),
+  INTEGRAL_SIGNED = static_cast<int32_t>(type_group_id::INTEGRAL_SIGNED),
+  FLOAT           = static_cast<int32_t>(type_group_id::FLOATING_POINT),
+  DECIMAL         = static_cast<int32_t>(type_group_id::FIXED_POINT),
+  TIMESTAMP       = static_cast<int32_t>(type_group_id::TIMESTAMP),
+  STRING          = static_cast<int32_t>(cudf::type_id::STRING),
+  LIST            = static_cast<int32_t>(cudf::type_id::LIST),
+  STRUCT          = static_cast<int32_t>(cudf::type_id::STRUCT)
+};
```
Comment (ttnghia) [inline cpp/benchmarks/io/nvbench_helpers.hpp:36]: Why don't use enums from `cudf::type_id` directly?

Comment (ttnghia) [reply cpp/benchmarks/io/nvbench_helpers.hpp:36]: Can the types like `INTEGRAL` be just `type_id::INT32`?

Comment (vuule) [reply cpp/benchmarks/io/nvbench_helpers.hpp:36]: INTEGRAL represents all integral types. Types are used in round-robin order so we get a mix of columns with different, but related, types. This allows us to cover all supported types without having separate benchmarks for each type.

**Review by ttnghia — APPROVED** [review]

File: cpp/benchmarks/io/nvbench_helpers.hpp (hunk 2 above)
Comment (PointKernel) [reply cpp/benchmarks/io/nvbench_helpers.hpp:36]: > Why don't use enums from cudf::type_id directly?

As you might notice, `data_type` is a combination of `type_group_id` and `cudf::type_id` thus the wrapper layer is necessary.

**Comment by PointKernel** [comment]
rerun tests

**Comment by PointKernel** [comment]
@gpucibot merge

**Pull request merged**